src/Workspaces/Core/Portable/Storage/SQLite/Interop/SqlConnection.cs

```diff
@@ -99,7 +119,11 @@ internal void Close_OnlyForUseBySqlPersistentStorage()
             Contract.ThrowIfNull(_handle);
 
             // release all the cached statements we have.
-            foreach (var statement in _queryToStatement.Values)
+            //
+            // use the struct-enumerator of our dictionary to prevent any allocations here.  We
+            // don't want to risk an allocation causing an OOM which prevents executing the
+            // following cleanup code.
```

Comment:
📝 This is close to correct, but slightly off. `Dictionary<TKey, TValue>.Values` already returns `ValueCollection` which has a struct enumerator, but the allocation can still occur for the first access of `Values` (lazily initialized). The allocation we are avoiding is the `ValueCollection`.

Reply:
i'm not sure how to phrase this better.  this was hte allocation i was avoiding. a nd we avoid it by directly using the struct-enumerator of hte dictionary.